### PR TITLE
Give error when casacore is too old

### PR DESCRIPTION
**Pull request by tammojan**
The travis tests using KERN all fail on a unicode error. Probably this was fixed in casacore 3.1.1, which has unicode improvements. I'll bump to KERN5 to see if it helps, but KERN5 has casacore ~2.5.0~ 3.0.0, so probably not.

### CODE REVIEW

**Comment by tammojan** [comment]
I added an error to `setup.py` to raise an Exception when casacore is detected and the version can be extracted, but it is too old. The result is [here](https://travis-ci.org/github/casacore/python-casacore/jobs/675708518#L1297):
```
RuntimeError: Your casacore version is too old. Minimum is 3.1.1, you have 2.5.0
```

File: setup.py [hunk 1]
```diff
@@ -104,10 +104,11 @@ def find_casacore():
             casacoreversion = getCasacoreVersion()
         except:
             # getVersion was fixed in casacore 2.3.0
-            warnings.warn("Your casacore version is older than 2.3.0! You need to upgrade your casacore.")
```
Comment (gijzelaerr) [inline setup.py:107]: a setup script should never fail, so we should not raise an exception.

Comment (tammojan) [reply setup.py:107]: What should it do? It cannot logically succeed.

Comment (tammojan) [reply setup.py:107]: Ok, I'll print a warning instead.

**Comment by tammojan** [comment]
Rebased on master. I made an extension to `build_ext` that will crash if casacore is too old. This should make sure that setup.py can still be run to do e.g. `setup.py clean`. This is tested in one of the travis jobs.

**Comment by gijzelaerr** [comment]
hey! nog niet gereviewd :)

**Comment by tammojan** [comment]
Oops...

**Comment by gijzelaerr** [comment]
maar ziet er goed uit, mooi zerk. volgens mij zou dit wel moeten werken. laten we zien wat er gebeurt tijdens packagen.

**Pull request merged**